llvm/include/llvm/CodeGen/TargetPassConfig.h

```diff
@@ -496,6 +496,8 @@ class TargetPassConfig : public ImmutablePass {
 void registerCodeGenCallback(PassInstrumentationCallbacks &PIC,
                              LLVMTargetMachine &);
 
+bool usesGreedyOrDefaultRegisterAllocator();
```

Comment:
This doesn't work in general and "default" allocator is ambiguous. I'm -1 on anything that tries to expose the allocator in use to clang. I thought the result of the issue thread was fast RA would just start spilling everything?

Reply:
> I thought the result of the issue thread was fast RA would just start spilling everything?

And how do we know that we're in the fast register allocator before we reach the fast register allocator? Also, how does this not work in general?

Reply:
You don't, you just have fastRA do whatever horrible spilling it needs to make it work. Wasn't that what the referenced review in https://github.com/llvm/llvm-project/issues/20571#issuecomment-2061935279 was doing?

Reply:
+1 on what @arsenm said.
We shouldn't expose the allocator to upper layers.

Reply:
The problem is that code gen from the ISel DAG messes things up enough that "just spilling" the registers isn't easy at all and will require a heinous hack. But if that's what you want, fine.